geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/VMStatsMonitor.java

```diff
@@ -25,76 +26,63 @@
 /**
  * This class acts as a monitor and listen for VM stats update on behalf of MemberMBean.
  *
- *
  */
 public class VMStatsMonitor extends MBeanStatsMonitor {
-
-  private static final int VALUE_NOT_AVAILABLE = -1;
-
-  private volatile float cpuUsage = 0;
-
-  private static String processCPUTimeAttr = "ProcessCpuTime";
-
+  static final float VALUE_NOT_AVAILABLE = -1;
+  private static final String PROCESS_CPU_TIME_ATTRIBUTE = "ProcessCpuTime";
   private long lastSystemTime = 0;
-
   private long lastProcessCpuTime = 0;
+  private final boolean processCPUTimeAvailable;
+  private AtomicInteger cpuUsageBits = new AtomicInteger(Float.floatToIntBits(0f));
```

Comment:
Make "cpuUsageBits" final.
I think you could have just kept this one a volatile. It looks like the only reason for it even being volatile is to make sure that when you call getCpuUsage() it will return the most recent value.
It is okay to keep the atomic but the code that updates this one (refreshStats()) has other state and will compute bad results if multiple threads call refreshStats concurrently. I don't think that will ever happen. So just make this method synchronized. It will probably never be called concurrently so the synchronize will not hurt.

Reply:
👍